File: divkit2/ios/Sources/DivActionHandler.swift
import Foundation

protocol DivActionHandler {
    func handleAction(_ action: DivAction, context: DivContext) -> Bool
}

class DivFormActionHandler: DivActionHandler {
    func handleAction(_ action: DivAction, context: DivContext) -> Bool {
        if action.logId == "submit_form" {
            let formData = context.variables
            print("DivFormActionHandler: Submitting Form Data to \(String(describing: action.url)): \(formData)")
            // In a real app, perform network request here
            return true
        }
        return false
    }
}

class DivCompositeActionHandler: DivActionHandler {
    let handlers: [DivActionHandler]

    init(handlers: [DivActionHandler]) {
        self.handlers = handlers
    }

    func handleAction(_ action: DivAction, context: DivContext) -> Bool {
        for handler in handlers {
            if handler.handleAction(action, context: context) {
                return true
            }
        }
        return false
    }
}

File: divkit2/android/src/main/java/com/example/divkit2/DivActionHandler.kt
package com.example.divkit2

interface DivActionHandler {
    fun handleAction(action: DivAction, context: DivContext): Boolean
}

class DivFormActionHandler : DivActionHandler {
    override fun handleAction(action: DivAction, context: DivContext): Boolean {
        if (action.logId == "submit_form") {
            val formData = context.getAllVariables()
            println("DivFormActionHandler: Submitting Form Data to ${action.url}: $formData")
            // In a real app, perform network request here
            return true
        }
        return false
    }
}

class DivCompositeActionHandler(private val handlers: List<DivActionHandler>) : DivActionHandler {
    override fun handleAction(action: DivAction, context: DivContext): Boolean {
        for (handler in handlers) {
            if (handler.handleAction(action, context)) {
                return true
            }
        }
        return false
    }
}
